opm/parser/eclipse/EclipseState/Tables/SlgofTable.hpp

```diff
@@ -0,0 +1,78 @@
+/*
+  Copyright (C) 2015 by Andreas Lauser
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef OPM_PARSER_SLGOF_TABLE_HPP
+#define	OPM_PARSER_SLGOF_TABLE_HPP
+
+#include "SingleRecordTable.hpp"
+
+namespace Opm {
+    // forward declaration
+    class EclipseState;
+
+    class SlgofTable : protected SingleRecordTable {
+        typedef SingleRecordTable ParentType;
+
+        friend class EclipseState;
+
+        /*!
+         * \brief Read the SGOF keyword and provide some convenience
+         *        methods for it.
+         */
+        void init(Opm::DeckKeywordConstPtr keyword,
+                  int recordIdx)
+        {
+            ParentType::init(keyword,
+                             std::vector<std::string>{"SL", "KRG", "KROG", "PCOG"},
+                             recordIdx,
+                             /*firstEntityOffset=*/0);
+
+            ParentType::checkNonDefaultable("SL");
+            ParentType::checkMonotonic("SL", /*isAscending=*/true);
+            ParentType::applyDefaultsLinear("KRG");
```

Comment:
right. fixed.

> Shouldn't we add a test for this?

Ok, I added one. Though I don't trust it too much, because I think that this keyword an example where you make your test match the implementation, not vice-versa...